hail/python/hailtop/batch/backend.py

```diff
@@ -160,7 +157,12 @@ def __init__(self,
             flags += f' -v {gsa_key_file}:/gsa-key/key.json'
 
         self._extra_docker_run_flags = flags
-        self.__fs = RouterAsyncFS()
+        self.__requester_pays_fses = gcs_requester_pays_fs_cache(fs_constructor=RouterAsyncFS)
```

Comment:
Instead of having this field in both subclasses and requiring an abstract property, you can pass a `GCSRequesterPaysFSCache` to `super().__init__`

Reply:
good call, updated!